### ISSUE TITLE
"az eventgrid event-subscription create" does not support Service Bus Topic endpoint type but Azure Portal does

### ISSUE THREAD

**Issue by nwohlgemuth** [Azure/azure-cli#13579]
**Describe the bug**
In the azure portal, you can create an Azure Event Grid Subscription with a Service Bus Topic endpoint. However, the "az eventgrid event-subscription create" command does not support Service Bus Topic for the "--endpoint-type" argument.

**To Reproduce**
Run "az eventgrid event-subscription create --help" and see that the options listed for "--endpoint-type" do not include Service Bus Topic.

**Expected behavior**
That the options available in Azure portal are implemented in the Azure CLI.

**Environment summary**
Azure CLI version 2.6.0 on Linux

**Comment by yungezz** [comment]
routing to appropriate team

**Comment by lilyjma** [comment]
@jfggdl do we have any update on this?

**Comment by ahamad-MS** [comment]
Hello @nwohlgemuth nwohlgemuth. Thanks for reporting this issue. You are using quite old az cli version. Support service bus topic was already added to CLI long time ago. Please move to latest version (2.12 currently). The following is part of the create cmd help for event subscription


  --endpoint-type                                : The type of the destination endpoint.  Allowed
                                                     values: azurefunction, eventhub,
                                                     hybridconnection, servicebusqueue,
                                                     **### _servicebustopic_**, storagequeue, webhook.
                                                     Default: webhook.

$ az version
{
  "azure-cli": "2.12.0",
  "azure-cli-core": "2.12.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "ai-examples": "0.2.3"
  }
}

Thanks

**Comment by nwohlgemuth** [comment]
@ahamad-MS I opened this issue on May 19th. Azure CLI version 2.6.0 was released on May 19th. At that time, that was the newest version I could have possibly been using (same day as it was released!) but it did not support the requested feature. Since then, the feature was added. To workaround it in the meantime, I wrote code that directly made an HTTP request to the Azure API. Now that Azure CLI supports it, I will remove that code I wrote and use the Azure CLI.
